ee/clickhouse/queries/test/test_lifecycle.py

```diff
@@ -70,13 +76,53 @@ def test_test_account_filters_with_groups(self):
             self.team,
         )
 
-        self.assertEqual(sorted(res["status"] for res in result), ["dormant", "new", "resurrecting", "returning"])
-        for res in result:
-            if res["status"] == "dormant":
-                self.assertEqual(res["data"], [0, -1, 0, 0, -1, 0, 0, 0])
-            elif res["status"] == "returning":
-                self.assertEqual(res["data"], [0, 0, 0, 0, 0, 0, 0, 0])
-            elif res["status"] == "resurrecting":
-                self.assertEqual(res["data"], [1, 0, 0, 1, 0, 0, 0, 0])
-            elif res["status"] == "new":
-                self.assertEqual(res["data"], [0, 0, 0, 0, 0, 0, 0, 0])
+        self.assertLifecycleResults(
+            result,
+            [
+                {"status": "dormant", "data": [0, -1, 0, 0, -1, 0, 0, 0]},
+                {"status": "new", "data": [0, 0, 0, 0, 0, 0, 0, 0]},
+                {"status": "resurrecting", "data": [1, 0, 0, 1, 0, 0, 0, 0]},
+                {"status": "returning", "data": [0, 0, 0, 0, 0, 0, 0, 0]},
+            ],
+        )
+
+    @snapshot_clickhouse_queries
+    def test_lifecycle_edge_cases(self):
+        # This test tests behavior when created_at is different from first matching event and dormant/resurrecting/returning logic
```

Comment:
What specific edge case is this? Could we spell out explicitly? I'm guessing it's specifically that there will be no NEW status? Are there any other cases we're testing for here?

Reply:
There's the NEW case mainly. But also checked the definitions of dormant/resurrecting/returning as I don't have full confidence in prior tests.

Can drop those if you think they're unneeded.

Reply:
I would generally go for explicit and minimal to make if super obvious the behaviour I was checking, but that's personal preference so not a blocker